src/net.cpp

```diff
@@ -2814,7 +2814,15 @@ bool CConnman::DisconnectNode(const CSubNet& subnet)
 
 bool CConnman::DisconnectNode(const CNetAddr& addr)
 {
-    return DisconnectNode(CSubNet(addr));
+    bool disconnected = false;
```

Comment:
Is this change still needed with #20852? It seems to me that makes the old construction work?

Reply:
No, if #20852 is merged, then this PR is not needed, see https://github.com/bitcoin/bitcoin/pull/20849#issuecomment-754122796.

I first opened this PR to fix `DisconnectNode()` and then realized that the same problem exists in `BanMan`, thus opened #20852 which fixes both by modifying `CSubNet`.

Maybe this PR could be preferred as less risky compared to #20852. It is ok if this is merged first and later #20852 is also merged - they do not conflict and both go in `master`.